api/views/share.py

```diff
@@ -53,9 +55,24 @@ def get_object(self):
 
         return obj
 
+    # Override to filter out deleted items
+    def list(self, request, *args, **kwargs):
+        queryset = self.filter_queryset(self.get_queryset())
+        try:
+            queryset = queryset.filter(is_deleted=False)
+        except:
```

Comment:
Bare except blocks are very dangerous. This will catch and swallow any exception, included KeyboardInterrupts.

Catch the `FieldError` that would be raised here.